### PR TITLE
Add BertLarge to FP16 / Torch Benchmarks

### PR DESCRIPTION
**Pull request by reichlfl**
This adds BertLarge fp16 PyTorch BMs in various batch sizes.

The model is part of our current CUDA Focus Suite: https://github.com/openxla/iree/issues/12873

### CODE REVIEW

File: build_tools/python/e2e_test_framework/models/model_groups.py [hunk 1]
```diff
@@ -136,9 +148,10 @@
     torch_models.MODEL_CLIP_TEXT_SEQLEN64_FP32_TORCH,
     torch_models.MODEL_UNET_2D_FP32_TORCH,
     torch_models.EFFICIENTNET_B7_FP32_TORCH,
-]
+] + BERT_LARGE_FP16_TORCH_BATCHES
```
Comment (pzread) [inline build_tools/python/e2e_test_framework/models/model_groups.py]: Do we need to benchmark all batch sizes in the default benchmark suite? Maybe just one batch size and the rest can go into `CUDA_MODELS_LONG`?

Comment (mariecwhite) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: +1 on this. However you might run into an issue where the same model cannot be used in `CUDA_MODELS` and `CUDA_MODELS_LONG`. I believe @pzread is working on a fix.

Comment (reichlfl) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: SGTM, I'll move all but one of the batch sizes to CUDA_LONG.

Comment (reichlfl) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: Done

File: build_tools/python/e2e_test_framework/models/torch_models.py [hunk 2]
```diff
@@ -166,6 +166,97 @@
     entry_function="forward",
     input_types=["1280x384xi64", "1280x384xi64"])
 
+# FP16 Versions
+BERT_LARGE_1X384_FP16_TORCH = common_definitions.Model(
```
Comment (pzread) [inline build_tools/python/e2e_test_framework/models/torch_models.py:170]: (Optional): I think we can actually write code to generate configs for different batch sizes as they are identical in many places. Just suggestions, don't need to do it in this PR : )

**Review by pzread — APPROVED** [review]

File: build_tools/python/e2e_test_framework/models/model_groups.py (hunk 1 above)
Comment (pzread) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: Yes, support benchmark in multiple presets in tracked on #13683

**Pull request merged**